### PR TITLE
Added error for unauthed active_storage

### LINKED ISSUE #253: Authenticate ActiveStorage controllers by default
According to https://edgeguides.rubyonrails.org/active_storage_overview.html#authenticated-controllers:

> All Active Storage controllers are publicly accessible by default. The generated URLs use a plain signed_id, making them hard to guess but permanent. Anyone that knows the blob URL will be able to access it, even if a before_action in your ApplicationController would otherwise require a login.

This was a surprise to me, since we generally authenticate the record quite well, but then go on to do something like `record.file.url` in our view, generating a URL that is permanent and unauthenticated. It also operates independently of any expiry set in whatever storage service we are using, since it goes through a redirect controller in ActiveStorage.

I think the easiest way to make this default to something reasonably safe would be to patch the ActiveStorage controllers to authenticate a signed-in user. Alternatively, we can disable these controllers altogether and require applications that need to serve files in this way to provide their own implementations that suits the type of access they need to support.

### PR DESCRIPTION
**Pull request by Breccan**
Fixes #253 

Discussion pointed towards forcing the choice rather than implementing a particular direction by default.

### CODE REVIEW

**Comment by dp4rk** [comment]
You will need to modify .github/workflows/ci.yml on the `basic-with-skips` step to not skip active storage now.

File: app/controllers/active_storage/base_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+# The base class for all Active Storage controllers. 
+# Copied from 
+# https://github.com/rails/rails/blob/main/activestorage/app/controllers/active_storage/base_controller.rb
+class ActiveStorage::BaseController < ActionController::Base # rubocop:disable Rails/ApplicationController
+  include ActiveStorage::SetCurrent
+
+  protect_from_forgery with: :exception
+
+	before_action :authenticated?
```
Comment (joshmcarthur) [inline app/controllers/active_storage/base_controller.rb:11]: got some inconsistent indentation here

File: app/controllers/active_storage/base_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  self.etag_with_template_digest = false
+
+	private
+	
+	# ActiveStorage defaults to security via obscurity approach to serving links
+	# If this is acceptable for you ruse case then this authenticable test can be
```
Comment (joshmcarthur) [inline app/controllers/active_storage/base_controller.rb:18]: ruh-roh, `ruse case`

File: app/controllers/active_storage/base_controller.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	# removed. If not then code should be added to only serve files appropriately.	
+  # https://edgeguides.rubyonrails.org/active_storage_overview.html#proxy-mode
+	def authenticated?
```
Comment (joshmcarthur) [inline app/controllers/active_storage/base_controller.rb:21]: Are there common implementations we can include to refer to?

**Review by joshmcarthur — CHANGES_REQUESTED** [review]

File: app/controllers/active_storage/base_controller.rb (hunk 3 above)
Comment (eoinkelly) [reply app/controllers/active_storage/base_controller.rb:21]: I'm not aware of any open source ones.

**Comment by eoinkelly** [comment]
Knowing how busy @Breccan is, I've fixed up the formatting and merge issues here.

**Comment by eoinkelly** [comment]
We reviewed this in person and it's 👍

**Pull request merged**